src/angular/form-elements/dropdown/dropdown.component.ts

```diff
@@ -1,28 +1,224 @@
+     * @param value
+     */
+    private updateSelected(label: string, value: string, index?: number){
+        this.selectedLabel = label;
+        this.value = value;
+        this.baseEmitter.next(this.value);
+        this.selectedIndex = index || this.selectedIndex;
+        this.validateDropDown();
+    }
+
+    /**
+     * Get the value of selected drop-down group item
+     * @param dropdownGroup
+     * @returns {string}
+     */
+    public getDropDownGroupSelectedValue(dropdownGroup: IDropDownGroup):string{
+        if(this.selectedDropDownGroup && this.selectedDropDownGroup.title === dropdownGroup.title){
+            return this.value;
+        }
+        return null;
+    }
+
+    /**
+     * Check if our options are IDropDownGroup items
+     */
+    public isDropDownGroupList(): boolean{
+        return this.options && this.options.length && this.options[0].hasOwnProperty('title') || false;
+    }
+
+    /**
+     * Check if our options are IDropDownItem items
+     */
+    private isDropDownItemList(): boolean{
+        return this.options && this.options.length && this.options[0].hasOwnProperty('label') || false
+    }
+
+    /**
+     * Get the label of the selected option
+     */
+    public getSelectedLabel(): string{
+       return this.selectedLabel || null;
+    }
+
+    /**
+     * Toggle show/hide drop-down list
+     */
+    public toggleDropdown(){
+        if(!this.disabled){
+            this.show = !this.show;
+        }
+    }
+
+    /**
+     * When users clicks outside the drop-down it will be closed
+     */
+    public onClickOutside(event){
+        if(!event.target.classList.contains('js-sdc-dropdown--toggle-hook')){
+            this.show = false;
+        }
+    }
+
+    private setValueIfExistOnGroupList(value: string): void{
+        const selectedDropDownGroup = this.options.find((dropDownGroup,index)=>{
+            const checkDropDownGroup = <IDropDownGroup>dropDownGroup;
+            const callback = this.setFoundValueCallback.bind(this, true, value);
+            return !!checkDropDownGroup.items.find(callback);
+        });
+        if(selectedDropDownGroup){
+            this.selectedDropDownGroup = <IDropDownGroup>selectedDropDownGroup;
+        }
+    }
+
+   private setFoundValueCallback(isItem: boolean, value: string, option: any, index:number){
+        const optionValue = isItem?(<IDropDownItem>option).value:option;
+        if(optionValue === value){
+            const optionLabel = isItem?(<IDropDownItem>option).label:option;
+            this.updateSelected(optionLabel, optionValue, index);
+            return true;
+        }
+        return false;
+   }
+
+    /**
+     * Test if the specified value exist on the option list, if it is,
+     * add it as the enw value
+     * @param value
+     */
+    private setValueIfExist(value:string): void{
+        if(this.isDropDownGroupList()){
+            this.setValueIfExistOnGroupList(value);
+        }else if(this.isDropDownItemList()){
```

Comment:
Better do this check inside the function .setFoundValueCallback() instead of passing true/false as a first argument dependent on this check.